### ISSUE TITLE
[STM32F4Discovery]: unknown chip id! 0x413

### ISSUE THREAD

**Issue by Catherinecyj** [stlink-org/stlink#1241]
- [x] Programmer/board type: Stlink /V2-onboard
- [x] Programmer firmware version: V2J37S26
- [x] Operating system: Linux (Ubuntu)
- [x] Stlink tools version and/or git commit hash: 1.7.0-186-gc4762e6
- [x] Stlink commandline tool name: st-flash, st-info
- [x] Target chip (and optional board): STM32F407VGT6U (STM32F4Discovery)

Commandline output:
```
st-flash write testvectors.bin 0x8000000
st-flash 1.7.0-186-gc4762e6
/usr/local/stlink/chip: No such file or directory
2022-04-09 WARN common.c: unknown chip id! 0x413
Failed to connect to target
```

Commandline output:
```
st-info --probe
/usr/local/stlink/chip: No such file or directory
Found 1 stlink programmers
    version: V2J37S26
    serial:    066DFF485157808667044350
    flash:     0
    sram:     0
    chipid:   0x413
```
Expected/description:
It seems like the device ID is 0x413 exactly, for  I tried to use STM32 Cube Programmer and can recognized the device and the Device ID shown:
![image](https://user-images.githubusercontent.com/100463326/162553859-0d7e75dd-f37a-49f3-88dd-149e5c214be2.png)
Also can program the .bin file successfully.

Catherine

**Comment by Nightwalker-87** [comment]
Please use the issue template in the future - this ticket is actually invalid according to our project guidelines.

Duplicate of #1228.

**Comment by bsalha1** [comment]
I am also getting this on my STM32F407VET6

**Comment by Nightwalker-87** [comment]
This is not a chip-specific issue.
The output message `/usr/local/stlink/chip: No such file or directory` indicates a compilation bug regarding the installation path of the .chip configuration files used to detect MCU devices.

This bug has been fixed in commit f93adb92f2e4ecf05a9361cb723c98693586929d.
The correct installation path is `/usr/share/stlink/chips`.
In order to resolve this, please clean-up your installation with `make uninstall`, then `make clean`, preferably check out the latest commit on the `develop` branch and proceed with `make install`.
